.github/workflows/run-tests.yaml

```
@@ -24,3 +27,33 @@ jobs:
                --env BOARD_KEY=${{ secrets.BOARD_KEY }}
                todo-app:test
                tests/e2e/test_chromium.py
+
+  deploy:
+    name: Deploy into production
+    needs: run-tests
+    if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+    runs-on: ubuntu-latest
+    env:
+      HEROKU_API_KEY: "${{ secrets.HEROKU_API_KEY }}"
+      DOCKER_USERNAME: "${{ secrets.DOCKER_USERNAME }}"
+      DOCKER_PASSWORD: "${{ secrets.DOCKER_PASSWORD }}"
+    steps:
+    - uses: actions/checkout@v1
+    - name: Login to Docker Hub
+      run: echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin
+    - name: Build production docker image
+      run: docker build --target production --tag $DOCKER_USERNAME/todo-app-production:${GITHUB_SHA::6} --tag $DOCKER_USERNAME/todo-app-production:latest .
+    - name: Push commit hash tagged build to Docker Hub
+      run: docker push $DOCKER_USERNAME/todo-app-production:${GITHUB_SHA::6}
+    - name: Push latest tagged build to Docker Hub
+      run: docker push $DOCKER_USERNAME/todo-app-production:latest
```

Very minor, but you can combine these by asking docker to use the `--all-tags` flag so that it pushes both of these at once:
```suggestion
    - name: Push tagged builds to Docker Hub
      run: docker push $DOCKER_USERNAME/todo-app-production:${GITHUB_SHA::6} --all-tags
```